Sources/Architecture/ARM/SoftwareBreakpointManager.cpp

```diff
@@ -35,7 +35,10 @@ void SoftwareBreakpointManager::clear() {
 }
 
 ErrorCode SoftwareBreakpointManager::add(Address const &address, Type type,
-                                         size_t size) {
+                                         size_t size, Mode mode) {
+  if (mode != kModeExec)
```

Comment:
You can move this check in `super::add()` and avoid having to duplicate it in the x86 counterpart of this file.